src/components/CourseFilter/CourseFilter.tsx

```diff
@@ -0,0 +1,192 @@
+'use client';
+
+import React, { useEffect, useState } from 'react';
+import {
+  Box,
+  Autocomplete,
+  TextField,
+  Button,
+  Typography,
+  FormControl,
+  ButtonGroup,
+} from '@mui/material';
+import { useTheme } from '@mui/material/styles';
+import { Key, useCallback } from 'react';
+import { CourseWithTagsAndStudentCount } from '@/lib/prisma/courses';
+import { usePathname, useSearchParams, useRouter } from 'next/navigation';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+
+type CourseFilterProps = {
+  initialCourses: CourseWithTagsAndStudentCount[];
+  initialTags: any;
+};
```

Comment:
```suggestion
import { Tag } from '@prisma/client';

type CourseFilterProps = {
  initialCourses: CourseWithTagsAndStudentCount[];
  initialTags: Tag[];
};
```

Can swap the any to a Tag array.